web/src/containers/ApdApplication.js

```diff
@@ -16,50 +16,101 @@ import { selectApdOnLoad } from '../actions/apd';
 import StateProfile from '../components/ApdStateProfile';
 import ProposedBudget from '../components/ProposedBudget';
 
-const ApdApplication = ({
-  apdSelected,
-  place,
-  selectApdOnLoad: dispatchSelectApdOnLoad
-}) => {
-  if (!apdSelected) {
-    dispatchSelectApdOnLoad('/apd');
-    return <Redirect to="/" />;
+const unsavedPrompt =
+  'You have unsaved changes to your APD. Are you sure you want to change pages?';
```

Comment:
You have unsaved changes to your APD. Do you want to leave this page without saving?

"Are you sure" prompts have a glaze-over effect. People see them too much and burying the action behind them sometimes leads people to not think it through.